docs/create/define_a_subquery.md

```diff
@@ -45,12 +47,50 @@ The `network.types` declare the specific types supported by this blockchain, als
 - The dataSources defines the data will be extracted and the logic of data transformation to be applied. 
   - For `dataSources.kind`, we are only supporting `substrate/Runtime` at the moment.
   - The `startBlock` specify the block height to start indexing from.
+  - The `dataSources.filter` will filtering the data source to execute by the network endpoint spec name, see [network filter](#network-filter )
   - In `mapping.handlers` will list all the [mapping functions](#mapping-function) and their corresponding handler types,
-  also [filters](#apply-filter).
-  
-  
-### Apply filter
+  also [filters](#mapping-filter).
+
+
+### Network filter  
+
+Commonly, the user will create a subquery and expect to (re)use it for their testnet and mainnet (e.g Polkadot and Kusama), and for the different network, their start indexing block is likely to vary. Therefore, we are demanding to run different datasource for different network.
```

Comment:
`demanding` seems too strong. We just provide another option. The old way also works and for user don't run into this issue, we still want them to use the old way.